### ISSUE TITLE
Exception Handling with 3.4.1

### ISSUE THREAD

**Issue by mtraynham** [kotest/kotest#1107]
On 3.4.0, a NullPointer exception would just fail the particular test case that encountered the exception.  By upgrading to `io.kotlintest:kotlintest-runner-junit5:3.4.1`, a NullPointer now dumps a pretty nasty stack trace with Gradle and it seems to mark the test case as failed twice, with one of the errors being `com.me.test.TestSpec > com.me.test.TestSpec > executionError FAILED`.

----
On 3.4.0, you would see something like:
```
com.me.test.TestSpec > It: Can do something FAILED
    java.lang.AssertionError: Expected null to not be null.
        at com.nhaarman.expect.FailKt.fail(Fail.kt:27)
        at com.nhaarman.expect.Matcher.toNotBeNull(Matcher.kt:51)
        at com.nhaarman.expect.Matcher.toNotBeNull$default(Matcher.kt:49)
        at com.me.test.TestSpec$1$1$2.invokeSuspend(TestSpec$1$1$2.kt:101)
        at com.me.test.TestSpec$1$1$2.invoke(TestSpec$1$1$2.kt)
```
And at the end of the Gradle test run:
```
Gradle Test Executor 17 finished executing tests.

> Task :project:test FAILED

13 tests completed, 1 failed
Finished generating test XML results (0.004 secs) into: /project/build/test-results/test
Generating HTML test report...
Finished generating test html results (0.006 secs) into: /project/build/reports/tests/test
:project:test (Thread[Execution worker for ':',5,main]) completed. Took 31.101 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':project:test'.
> There were failing tests. See the report at: file:///project/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 50s
14 actionable tasks: 8 executed, 6 up-to-date
```

----
On 3.4.1, you see something like:
```
com.me.test.TestSpec > com.me.test.TestSpec > executionError FAILED
    java.lang.AssertionError: Expected null to not be null.
        at com.nhaarman.expect.FailKt.fail(Fail.kt:27)
        at com.nhaarman.expect.Matcher.toNotBeNull(Matcher.kt:51)
        at com.nhaarman.expect.Matcher.toNotBeNull$default(Matcher.kt:49)
        at com.me.test.TestSpec$1$1$2.invokeSuspend(TestSpec.kt:101)
        at com.me.test.TestSpec$1$1$2.invoke(TestSpec.kt)
null
java.lang.NullPointerException
        at org.gradle.api.internal.tasks.testing.results.StateTrackingTestResultProcessor.started(StateTrackingTestResultProcessor.java:45)
        at org.gradle.api.internal.tasks.testing.results.AttachParentTestResultProcessor.started(AttachParentTestResultProcessor.java:38)
        at jdk.internal.reflect.GeneratedMethodAccessor176.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.FailureHandlingDispatch.dispatch(FailureHandlingDispatch.java:30)
        at org.gradle.internal.dispatch.AsyncDispatch.dispatchMessages(AsyncDispatch.java:87)
        at org.gradle.internal.dispatch.AsyncDispatch.access$000(AsyncDispatch.java:36)
        at org.gradle.internal.dispatch.AsyncDispatch$1.run(AsyncDispatch.java:71)
        at org.gradle.internal.concurrent.InterruptibleRunnable.run(InterruptibleRunnable.java:42)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:830)
Received a failure event for test with unknown id '18.20'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
java.lang.IllegalArgumentException: Received a failure event for test with unknown id '18.20'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
        at org.gradle.api.internal.tasks.testing.results.StateTrackingTestResultProcessor.failure(StateTrackingTestResultProcessor.java:103)
        at org.gradle.api.internal.tasks.testing.results.AttachParentTestResultProcessor.failure(AttachParentTestResultProcessor.java:43)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.FailureHandlingDispatch.dispatch(FailureHandlingDispatch.java:30)
        at org.gradle.internal.dispatch.AsyncDispatch.dispatchMessages(AsyncDispatch.java:87)
        at org.gradle.internal.dispatch.AsyncDispatch.access$000(AsyncDispatch.java:36)
        at org.gradle.internal.dispatch.AsyncDispatch$1.run(AsyncDispatch.java:71)
        at org.gradle.internal.concurrent.InterruptibleRunnable.run(InterruptibleRunnable.java:42)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:830)
Received a completed event for test with unknown id '18.20'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
java.lang.IllegalArgumentException: Received a completed event for test with unknown id '18.20'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
        at org.gradle.api.internal.tasks.testing.results.StateTrackingTestResultProcessor.completed(StateTrackingTestResultProcessor.java:79)
        at org.gradle.api.internal.tasks.testing.results.AttachParentTestResultProcessor.completed(AttachParentTestResultProcessor.java:56)
        at jdk.internal.reflect.GeneratedMethodAccessor180.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.FailureHandlingDispatch.dispatch(FailureHandlingDispatch.java:30)
        at org.gradle.internal.dispatch.AsyncDispatch.dispatchMessages(AsyncDispatch.java:87)
        at org.gradle.internal.dispatch.AsyncDispatch.access$000(AsyncDispatch.java:36)
        at org.gradle.internal.dispatch.AsyncDispatch$1.run(AsyncDispatch.java:71)
        at org.gradle.internal.concurrent.InterruptibleRunnable.run(InterruptibleRunnable.java:42)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:830)
Received a completed event for test with unknown id '18.19'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
java.lang.IllegalArgumentException: Received a completed event for test with unknown id '18.19'. Registered test ids: '[18.16, :project:test, 18.1, 18.6, 18.7, 18.5, 18.10, 18.11, 18.8, 18.9, 18.14, 18.15, 18.12, 18.13]'
        at org.gradle.api.internal.tasks.testing.results.StateTrackingTestResultProcessor.completed(StateTrackingTestResultProcessor.java:79)
        at org.gradle.api.internal.tasks.testing.results.AttachParentTestResultProcessor.completed(AttachParentTestResultProcessor.java:56)
        at jdk.internal.reflect.GeneratedMethodAccessor180.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.FailureHandlingDispatch.dispatch(FailureHandlingDispatch.java:30)
        at org.gradle.internal.dispatch.AsyncDispatch.dispatchMessages(AsyncDispatch.java:87)
        at org.gradle.internal.dispatch.AsyncDispatch.access$000(AsyncDispatch.java:36)
        at org.gradle.internal.dispatch.AsyncDispatch$1.run(AsyncDispatch.java:71)
        at org.gradle.internal.concurrent.InterruptibleRunnable.run(InterruptibleRunnable.java:42)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:830)

com.me.test.TestSpec > It: Can do something FAILED
    java.lang.AssertionError: Expected null to not be null.
        at com.nhaarman.expect.FailKt.fail(Fail.kt:27)
        at com.nhaarman.expect.Matcher.toNotBeNull(Matcher.kt:51)
        at com.nhaarman.expect.Matcher.toNotBeNull$default(Matcher.kt:49)
        at com.me.test.TestSpec$1$1$2.invokeSuspend(TestSpec.kt:101)
        at com.me.test.TestSpec$1$1$2.invoke(TestSpec.kt)
```

At the end of the Gradle run, this also seems to prevent generating the HTML report

```
Gradle Test Executor 18 finished executing tests.

> Task :project:test FAILED

14 tests completed, 2 failed
:project:test (Thread[Execution worker for ':' Thread 4,5,main]) completed. Took 29.467 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':project:test'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 39s
14 actionable tasks: 3 executed, 11 up-to-date
```

----
Here is a smaller test case that exposes the issue:
```
package com.me.test

import com.nhaarman.expect.expect
import io.kotlintest.specs.DescribeSpec

class TestSpec : DescribeSpec() {
    init {
        describe("Foo") {
            it("works") {
                val foo: String? = null
                expect(foo).toNotBeNull()
            }
        }
    }
}
```

**Comment by sksamuel** [comment]
I've confirmed that this is fixed in master for 4.0 and I would also suggest you upgrade to 3.4.2

**Comment by buckler4** [comment]
I am seeing a similar error with Kotest 4.1.0.293-SNAPSHOT (or 4.0.3) running with either Kotlin 1.3.70 or 1.4-M1.  Thanks,

Tested with similar code in StringSpec:
```
 "Foo works" {
            val foo: String? = null
            foo shouldNotBe null
    }
```

```
<null> should not equal <null>
java.lang.AssertionError: <null> should not equal <null>
	at biokotlin.data.CodonTableTest$1$7.invokeSuspend(CodonTableTest.kt:71)
	at biokotlin.data.CodonTableTest$1$7.invoke(CodonTableTest.kt)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invokeSuspend(executions.kt:16)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithGlobalAssert(executions.kt:43)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invokeSuspend(executions.kt:15)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invokeSuspend(executions.kt:29)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invoke(executions.kt)
	at io.kotest.core.test.AssertionModeKt.executeWithAssertionsCheck(AssertionMode.kt:39)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithAssertionModeCheck(executions.kt:28)
	at io.kotest.core.runtime.ExecutionsKt.executeWithBehaviours(executions.kt:14)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invokeSuspend(TestExecutor.kt:182)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:171)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ExecutorExecutionContext$executeWithTimeoutInterruption$$inlined$suspendCoroutine$lambda$2.invokeSuspend(ExecutorExecutionContext.kt:47)
	(Coroutine boundary)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:175)
	at io.kotest.core.runtime.TestExecutor.executeAndWait-xkB6VbI(TestExecutor.kt:168)
	at io.kotest.core.runtime.TestExecutor.invokeTestCase(TestExecutor.kt:150)
	at io.kotest.core.runtime.TestExecutor.executeActiveTest(TestExecutor.kt:117)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invokeSuspend(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.executeIfActive(TestExecutor.kt:86)
	at io.kotest.core.runtime.TestExecutor.intercept(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.execute(TestExecutor.kt:55)
	at io.kotest.core.engine.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:62)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invokeSuspend(SingleInstanceSpecRunner.kt:73)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:79)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:111)
	at io.kotest.core.engine.SpecExecutor.runTests(SpecExecutor.kt:114)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:36)
Caused by: java.lang.AssertionError: <null> should not equal <null>
	at biokotlin.data.CodonTableTest$1$7.invokeSuspend(CodonTableTest.kt:71)
	at biokotlin.data.CodonTableTest$1$7.invoke(CodonTableTest.kt)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invokeSuspend(executions.kt:16)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithGlobalAssert(executions.kt:43)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invokeSuspend(executions.kt:15)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invokeSuspend(executions.kt:29)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invoke(executions.kt)
	at io.kotest.core.test.AssertionModeKt.executeWithAssertionsCheck(AssertionMode.kt:39)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithAssertionModeCheck(executions.kt:28)
	at io.kotest.core.runtime.ExecutionsKt.executeWithBehaviours(executions.kt:14)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invokeSuspend(TestExecutor.kt:182)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:171)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ExecutorExecutionContext$executeWithTimeoutInterruption$$inlined$suspendCoroutine$lambda$2.invokeSuspend(ExecutorExecutionContext.kt:47)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:272)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.runtime.ExecutorExecutionContext.executeWithTimeoutInterruption-D5N0EJY(ExecutorExecutionContext.kt:46)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2.invokeSuspend(TestExecutor.kt:170)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:102)
	at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:111)
	at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:32)
	at io.kotest.core.runtime.TestExecutor.executeAndWait-xkB6VbI(TestExecutor.kt:168)
	at io.kotest.core.runtime.TestExecutor.invokeTestCase(TestExecutor.kt:150)
	at io.kotest.core.runtime.TestExecutor.executeActiveTest(TestExecutor.kt:117)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invokeSuspend(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.TestExecutor.executeIfActive(TestExecutor.kt:86)
	at io.kotest.core.runtime.TestExecutor.intercept(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.execute(TestExecutor.kt:55)
	at io.kotest.core.engine.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:62)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invokeSuspend(SingleInstanceSpecRunner.kt:73)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:79)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invoke(SingleInstanceSpecRunner.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.engine.SingleInstanceSpecRunner.execute(SingleInstanceSpecRunner.kt:77)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:111)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invoke(SpecExecutor.kt)
	at io.kotest.core.engine.SpecExecutor.interceptSpec(SpecExecutor.kt:123)
	at io.kotest.core.engine.SpecExecutor.runTests(SpecExecutor.kt:114)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:36)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1$1.invokeSuspend(KotestEngine.kt:66)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:272)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1.run(KotestEngine.kt:65)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

biokotlin.data.CodonTableTest > Foo works FAILED
    java.lang.AssertionError at CodonTableTest.kt:71
        Caused by: java.lang.AssertionError at CodonTableTest.kt:71

<null> should not equal <null>
java.lang.AssertionError: <null> should not equal <null>
	at biokotlin.data.CodonTableTest$1$7.invokeSuspend(CodonTableTest.kt:71)
	at biokotlin.data.CodonTableTest$1$7.invoke(CodonTableTest.kt)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invokeSuspend(executions.kt:16)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithGlobalAssert(executions.kt:43)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invokeSuspend(executions.kt:15)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invokeSuspend(executions.kt:29)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invoke(executions.kt)
	at io.kotest.core.test.AssertionModeKt.executeWithAssertionsCheck(AssertionMode.kt:39)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithAssertionModeCheck(executions.kt:28)
	at io.kotest.core.runtime.ExecutionsKt.executeWithBehaviours(executions.kt:14)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invokeSuspend(TestExecutor.kt:182)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:171)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ExecutorExecutionContext$executeWithTimeoutInterruption$$inlined$suspendCoroutine$lambda$2.invokeSuspend(ExecutorExecutionContext.kt:47)
	(Coroutine boundary)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:175)
	at io.kotest.core.runtime.TestExecutor.executeAndWait-xkB6VbI(TestExecutor.kt:168)
	at io.kotest.core.runtime.TestExecutor.invokeTestCase(TestExecutor.kt:150)
	at io.kotest.core.runtime.TestExecutor.executeActiveTest(TestExecutor.kt:117)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invokeSuspend(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.executeIfActive(TestExecutor.kt:86)
	at io.kotest.core.runtime.TestExecutor.intercept(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.execute(TestExecutor.kt:55)
	at io.kotest.core.engine.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:62)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invokeSuspend(SingleInstanceSpecRunner.kt:73)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:79)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:111)
	at io.kotest.core.engine.SpecExecutor.runTests(SpecExecutor.kt:114)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:36)
Caused by: java.lang.AssertionError: <null> should not equal <null>
	at biokotlin.data.CodonTableTest$1$7.invokeSuspend(CodonTableTest.kt:71)
	at biokotlin.data.CodonTableTest$1$7.invoke(CodonTableTest.kt)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invokeSuspend(executions.kt:16)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2$1.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithGlobalAssert(executions.kt:43)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invokeSuspend(executions.kt:15)
	at io.kotest.core.runtime.ExecutionsKt$executeWithBehaviours$2.invoke(executions.kt)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invokeSuspend(executions.kt:29)
	at io.kotest.core.runtime.ExecutionsKt$wrapTestWithAssertionModeCheck$2.invoke(executions.kt)
	at io.kotest.core.test.AssertionModeKt.executeWithAssertionsCheck(AssertionMode.kt:39)
	at io.kotest.core.runtime.ExecutionsKt.wrapTestWithAssertionModeCheck(executions.kt:28)
	at io.kotest.core.runtime.ExecutionsKt.executeWithBehaviours(executions.kt:14)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invokeSuspend(TestExecutor.kt:182)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3$1.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invokeSuspend(TestExecutor.kt:176)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1$3.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ReplayKt.replay(replay.kt:19)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invokeSuspend(TestExecutor.kt:171)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2$1.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.ExecutorExecutionContext$executeWithTimeoutInterruption$$inlined$suspendCoroutine$lambda$2.invokeSuspend(ExecutorExecutionContext.kt:47)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:272)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.runtime.ExecutorExecutionContext.executeWithTimeoutInterruption-D5N0EJY(ExecutorExecutionContext.kt:46)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2.invokeSuspend(TestExecutor.kt:170)
	at io.kotest.core.runtime.TestExecutor$executeAndWait$2.invoke(TestExecutor.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:102)
	at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:111)
	at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:32)
	at io.kotest.core.runtime.TestExecutor.executeAndWait-xkB6VbI(TestExecutor.kt:168)
	at io.kotest.core.runtime.TestExecutor.invokeTestCase(TestExecutor.kt:150)
	at io.kotest.core.runtime.TestExecutor.executeActiveTest(TestExecutor.kt:117)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invokeSuspend(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.TestExecutor.executeIfActive(TestExecutor.kt:86)
	at io.kotest.core.runtime.TestExecutor.intercept(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.execute(TestExecutor.kt:55)
	at io.kotest.core.engine.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:62)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invokeSuspend(SingleInstanceSpecRunner.kt:73)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:79)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invoke(SingleInstanceSpecRunner.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.engine.SingleInstanceSpecRunner.execute(SingleInstanceSpecRunner.kt:77)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:111)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invoke(SpecExecutor.kt)
	at io.kotest.core.engine.SpecExecutor.interceptSpec(SpecExecutor.kt:123)
	at io.kotest.core.engine.SpecExecutor.runTests(SpecExecutor.kt:114)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:36)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1$1.invokeSuspend(KotestEngine.kt:66)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:272)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1.run(KotestEngine.kt:65)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by sksamuel** [comment]
@buckler4 that looks like it's working to me. What was the result you expected to see?

**Comment by buckler4** [comment]
@sksamuel - I expected to see the stack trace only once - but it looks like it is repeated.  I am new to KoTest so I may be missing something.  Thanks for the quick response.

**Comment by sksamuel** [comment]
Definitely happening in 4.x. I've made a new ticket to track.
